src/corefx/System.Globalization.Native/casing.cpp

```diff
@@ -20,46 +20,66 @@ extern "C" void ChangeCase(const UChar* lpSrc, int32_t cwSrcLength, UChar* lpDst
     int32_t dstIdx = 0;
 
     UBool isError = FALSE;
+    UChar32 srcCodepoint;
+    UChar32 dstCodepoint;
 
-    while (srcIdx < cwSrcLength)
-    {
-        UChar32 srcCodepoint;
-        UChar32 dstCodepoint;
-
-        // Decode the next one or two UTF-16 code units into a codepoint and update srcIdx to point to the next UTF-16 code unit to decode.
-        U16_NEXT(lpSrc, srcIdx, cwSrcLength, srcCodepoint);
+    // Iterate through the string, decoding the next one or two UTF-16 code units into a codepoint 
+    // and updating srcIdx to point to the next UTF-16 code unit to decode.  Then upper or lower
+    // case it, write dstCodepoint into lpDst at offset dstIdx, and update dstIdx.
+	// (The loop here has been manually cloned for each of the four cases, rather than having a 
```

Comment:
nit: indentation looks off by one space on some lines (or possibly a tab in there somewhere)